### PR TITLE
Added SDK and tests for checkConnectivity API

### PR DESCRIPTION
**Pull request by irrogozh**
## Description

1. New feature added to network SDK generated from https://github.com/Azure/azure-rest-api-specs/pull/1175/files

**Issue: https://github.com/Azure/azure-sdk-for-net/issues/3154**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [x] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 [assembly: AssemblyTitle("Microsoft Azure Network Management Library")]
 [assembly: AssemblyDescription("Provides Microsoft Azure Network management functions for managing the Microsoft Azure Network service.")]
 
-[assembly: AssemblyVersion("10.0.0.0")]
+[assembly: AssemblyVersion("10.1.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs]: @irrogozh please make similar updates to .csproj. Change in version in .csproj will what bump up the nuget package version.
Also delete all instances of project.json, global.json and *.xproj

File: src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
...
 [assembly: AssemblyFileVersion("10.1.0.0")]
 
 [assembly: AssemblyConfiguration("")]
```
Comment (shahabhijeet) [inline src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs]: please remove empty AssemblyConfiguration attributes. I know it has been there before, but if you are not going to use it, you can delete it.

File: src/SDKs/Network/Network.Tests/Tests/CheckConnectionTests.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+                string networkWatcherName = TestUtilities.GenerateName();
+                NetworkWatcher properties = new NetworkWatcher();
+                properties.Location = location;
+                
+                //Create network Watcher
+
+                ConnectivityParameters parameters = new ConnectivityParameters();
+                parameters.Source = new ConnectivitySource();
+                parameters.Source.ResourceId = "/subscriptions/6926fc75-ce7d-4c9e-a87f-c4e38c594eb5/resourceGroups/NwRgStage1/providers/Microsoft.Compute/virtualMachines/MultiTierApp0";
```
Comment (shahabhijeet) [inline src/SDKs/Network/Network.Tests/Tests/CheckConnectionTests.cs]: @irrogozh can all this information be obtained either from TestFramework or ResourceGroup that were created for the tests, e.g. TestEnvironment.KeyVaultPairs[ConnectionString.SubscriptionId]

File: src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+namespace Microsoft.Azure.Management.Network.Models
+{
+    using Azure;
+    using Management;
+    using Network;
+
+    /// <summary>
+    /// Defines values for IssueType.
+    /// </summary>
+    public static class IssueType
```
Comment (shahabhijeet) [inline src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs]: @irrogozh what is the purpose of these model types? Where are they being used?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (irrogozh) [inline src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs]: Done

File: src/SDKs/Network/Network.Tests/Tests/CheckConnectionTests.cs (hunk 3 above)
Comment (irrogozh) [inline src/SDKs/Network/Network.Tests/Tests/CheckConnectionTests.cs]: Not now, unfortunately. I'll fix it in the future.

File: src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs (hunk 4 above)
Comment (irrogozh) [inline src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs]: It's an enum for type in ConnectivityIssue.cs

Here is spec in swagger:
        "type": {
          "readOnly": true,
          "type": "string",
          "enum": [ "Unknown", "AgentStopped", "GuestFirewall", "DnsResolution", "SocketBind", "NetworkSecurityRule", "UserDefinedRoute", "PortThrottled", "Platform" ],
          "x-ms-enum": {
            "name": "IssueType",
            "modelAsString": true
          },
          "description": "The type of issue."
        },

File: src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (irrogozh) [inline src/SDKs/Network/Management.Network/Properties/AssemblyInfo.cs]: I'll keep it. Will coordinate with Ritwik Basu

File: src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs (hunk 4 above)
Comment (shahabhijeet) [inline src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs]: If this is part of the new feature, why aren't there any tests for these?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs (hunk 4 above)
Comment (irrogozh) [inline src/SDKs/Network/Management.Network/Generated/Models/IssueType.cs]: Because it doesn't work now. Failed it as an issue

**Comment by shahabhijeet** [comment]
@irrogozh please send a new PR for the remaining tests that need to be added for this new feature.
Making an exception to unblock.

**Pull request merged**